docker/Dockerfile

```diff
@@ -0,0 +1,156 @@
+FROM ubuntu:18.04 AS garage-18.04
+
+# http://bugs.python.org/issue19846
+# > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
+ENV LANG C.UTF-8
+
+# apt dependencies
+RUN \
+  apt-get -y -q update && \
+  # Prevents debconf from prompting for user input
+  # See https://github.com/phusion/baseimage-docker/issues/58
+  DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    # Dockerfile deps
+    wget \
+    unzip \
+    git \
+    curl \
+    # For building glfw
+    cmake \
+    xorg-dev \
+    # mujoco_py
+    # See https://github.com/openai/mujoco-py/blob/master/Dockerfile
+    # 18.04 repo is old, install glfw from source instead
+    # libglfw3 \
+    libglew-dev \
+    libosmesa6-dev \
+    patchelf \
+    # OpenAI baselines
+    libopenmpi-dev \
+    # virtualenv
+    python3 \
+    python3-pip \
+    python3-tk \
+    python3-virtualenv && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/*
+
+# Build GLFW because the Ubuntu 18.04 version is too old
+# See https://github.com/glfw/glfw/issues/1004
+RUN apt-get purge -y -v libglfw* && \
+  wget https://github.com/glfw/glfw/releases/download/3.3/glfw-3.3.zip && \
+  unzip glfw-3.3.zip && \
+  rm glfw-3.3.zip && \
+  cd glfw-3.3 && \
+  mkdir glfw-build && \
+  cd glfw-build && \
+  cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF .. && \
+  make -j"$(nproc)" && \
+  make install && \
+  cd ../../ && \
+  rm -rf glfw
+
+RUN groupadd -g 999 garage-user && \
```

Comment:
perhaps we can do something like this

```Dockerfile
ARG user=garage-user
ARG uid=999
RUN groupadd -g $uid $user && \
  useradd -m -r -u $uid -g $user $user && \
  chown -R $user:$user /home/$user
USER $user
ENV HOME /home/$user
WORKDIR $HOME
```

Then on the build command line (i.e. Makefile) we can do 
```sh
docker build --build_arg user="${USER}" --build_arg uid="${UID}" .
```
and the user/UID inside the container will match that of the building user. For people using the Dockerfile locally, this means files in `data/` will be created/owned by the user who built the dockerfile, which would cut down on weird permissions issues